stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/call/RtcSession.kt

```diff
@@ -849,15 +849,29 @@ public class RtcSession internal constructor(
         val tracks = participants.map { participant ->
             val trackDisplay = trackDisplayResolution[participant.sessionId] ?: emptyMap()
 
-            trackDisplay.entries.filter { it.value.visible }.map { display ->
-                dynascaleLogger.i { "[visibleTracks] $sessionId subscribing ${participant.sessionId} to : ${display.key}" }
-                TrackSubscriptionDetails(
-                    user_id = participant.user.value.id,
-                    track_type = display.key,
-                    dimension = display.value.dimensions,
-                    session_id = participant.sessionId
-                )
+            val tracks = mutableListOf<TrackSubscriptionDetails>()
+            val entries = trackDisplay.entries.filter { it.value.visible }
+            entries.forEach { entry ->
+                dynascaleLogger.i { "[visibleTracks] $sessionId subscribing ${participant.sessionId} to : ${entry.key}" }
+                if (participant.videoEnabled.value) {
+                    tracks.add(TrackSubscriptionDetails(
+                        user_id = participant.user.value.id,
+                        track_type = entry.key,
+                        dimension = entry.value.dimensions,
+                        session_id = participant.sessionId
+                    ))
+                }
+
+                if (participant.screenSharingEnabled.value) {
+                    tracks.add(TrackSubscriptionDetails(
+                        user_id = participant.user.value.id,
+                        track_type = TrackType.TRACK_TYPE_SCREEN_SHARE,
+                        dimension = entry.value.dimensions,
+                        session_id = participant.sessionId
+                    ))
+                }
```

Comment:
if the video is visible, but the screenshare isn't visible, this will make us subscribe to the screenshare won't it? 

also won't it add it to the tracks twice?

Reply:
AFAIK the tracks need to be added twice - in the bottom part of the screen you can still see the participant + at the top you see the screen share. So up to two tracks per participant.